thesis/03-implementation.tex

```diff
@@ -83,23 +83,491 @@ \subsubsection{Backend}
+
+\medskip
+
+To the right, there is a set of configuration sections with a decreasing level of specificity, correspondingly responsible for: \begin{description}
+	\item[edge] - Configuration of currently selected edges.
+	\item[step] - Setting a step name and a description.
+	\item[file] - Setting an Instruction name and description, visible after an Instruction is saved and processed in the system.
+\end{description}
+
+At the bottom of the Guide Creator, an interactive list of steps is present.
+
+\medskip
+
+\begin{figure}[H]
+  \caption{The process of creating an Instruction with the Guide Creator from a Designer's perspective.}
+  \label{3-designer-creator-flow}
+  \centering
+    \includegraphics[width=\textwidth]{assets/3-designer-creator-flow.png}
+\end{figure}
+
+Until \textit{Save the Guide} step, all data and actions are done on the client-side. When \textit{Save the Guide} action is invoked:
+\begin{itemize}
+	\item if a user is logged in, a request is sent to the Community with all configuration attached, or
+	\item a file download is initiated.
+\end{itemize}
+
+\begin{figure}[H]
+\caption{A Guide creation request. Almost all information is encoded in the \texttt{.fold} file}
+\begin{lstlisting}
+{
+	"private":false,
+	"guide_file":"data:text/json;base64,<base64 encoded guide>",
+	"thumbnail":"data:text/json;base64,<base64 thumbnail>"
+}
+\end{lstlisting}
+\end{figure}
+
+\begin{figure}[H]
+	\caption{A Guide creation response. The Instruction has not been processed yet, therefore, a \texttt{animation\_file} is not available.}
+	\label{guide-response}
+\begin{lstlisting}
+{
+	"id":6,
+	"owner":1,
+	"name":"penguin",
+	"published_at":"2020-12-01T18:49:18.131223Z",
+	"steps":9,
+	"guide_file":"https://api.<domain>/uploads/guides/file.fold",
+	"thumbnail_file":"https://api.<domain>/uploads/thumbnails/thumbnail.png",
+	"animation_file":null,
+	"status":"QUE",
+	"private":false,
+	"solved":false,
+	"liked":false,
+	"owner_username":"maciekmm1"
+}
+\end{lstlisting}
+\end{figure}
+
+
+\textbf{Guide Viewer}
+
+The purpose of the \textbf{Guide Viewer} is to show animated steps fluently. It consists of a model, a list of steps, and playback controls.
+
+\begin{figure}[H]
+  \caption{An overview of the Guide Viewer.}
+  \label{3-guide-viewer}
+  \centering
+    \includegraphics[width=\textwidth]{assets/3-guide-viewer.png}
+\end{figure}
+
+When a Guide is opened it fetches the \texttt{animation\_file} from the server and renders it.
+The response conforms to the same model as the aforementioned(\ref{guide-response}) guide creation.
+
+\medskip
+Playback controls allow rewinding, forwarding, stopping, and pausing to make it as approachable as possible for the Folder. Guide Viewer also provides functionality to load an arbitrary \tech{.fold} file by clicking the file icon located in the top-left corner of the screen.
+
+
+\subsubsection{Back-end}
```

Comment:
The section title is backend, but just right below it, we have a figure showing "community application", and then it's called "community component". Very confusing for the reader. I think we should have 1 convention and stick with it everywhere. 
I'd choose "Community backend" after all, and consequently use it anytime we refer to the backend part written in `Django`.

Reply:
All references now point to "Community backend" or "Solver backend". If something is related to both it's simply "Backend"